craft_providers/actions/snap_installer.py

```diff
@@ -189,6 +255,12 @@ def install_from_store(
             check=True,
             capture_output=True,
         )
+
+        InstanceConfiguration.update(
+            executor=executor,
+            data={"snaps": {snap_name: {"revision": store_revision}}},
```

Comment:
This `update` thing does not work like in dictionaries (took me several minutes of reading in pydantic's `main.py` code). It updates the first key, setting the rest.

I went forward and tested it:

```
(Pdb) InstanceConfiguration.update(executor=mock_executor, data={"snaps": {"some-snap-name": {"revision": 1234}}})
InstanceConfiguration(compatibility_tag=None, snaps={'some-snap-name': {'revision': 1234}})
(Pdb) InstanceConfiguration.update(executor=mock_executor, data={"snaps": {"other-snap-name": {"revision": 5678}}})
InstanceConfiguration(compatibility_tag=None, snaps={'other-snap-name': {'revision': 5678}})
```

This means that there's no point in having a "complex" structure below. And having a key called `snaps` is misleading because you cannot save more than one snap in this way.

Or am I missing something?

Reply:
Good catch. I've updated the function to recurse through nested data.  I've also added 2 more unit tests for this.